src/felzenszwalb.jl

```diff
@@ -9,73 +9,55 @@ The function returns the number of segments and index mapping from nodes of the
 Parameters:  
 -    edges          = Array of edges in RAG. Each edge is represented as `ImageEdge`.
 -    num_vertices   = Number of vertices in RAG
--    k              = Threshold for region splitting step. Larger threshold will result in bigger segments.
+-    k              = Threshold for region merging step. Larger threshold will result in bigger segments.
+-    min_size       = Minimum segment size 
 
 """
-function felzenszwalb_graph(edges::Array{ImageEdge}, num_vertices::Int, k::Real)
+function felzenszwalb_graph(edges::Array{ImageEdge}, num_vertices::Int, k::Real, min_size::Int = 0)
 
     num_edges = length(edges)
-    num_sets = num_vertices
-    rank = zeros(num_sets)
-    set_size = ones(num_sets)
-    parent = collect(1:num_sets)
+    G = IntDisjointSets(num_vertices)
+    set_size = ones(num_vertices)
     threshold = fill(k, num_vertices)
-    
-    function find_root(x)
-        x = Int(x)
-        if parent[x]!=x
-            parent[x] = find_root(parent[x])
-        end
-        return parent[x]
-    end
-
-    function merge_trees(x, y)
-        x = Int(x)
-        y = Int(y)
-        num_sets -= 1
-        if rank[x]<rank[y]
-            parent[x] = y
-            set_size[y] += set_size[x]
-            return y
-        elseif rank[x]>rank[y]
-            parent[y] = x
-            set_size[x] += set_size[y]
-            return x
-        else
-            parent[y] = x
-            set_size[x] += set_size[y]
-            rank[x] += 1
-            return x
-        end
-    end
 
```

Comment:
Nice with the `IntDijointSets` change! :100: for making it in DataStructures rather than here.